openedx/core/djangoapps/credit/tests/test_api.py

```diff
@@ -146,45 +173,220 @@ def test_requirements_to_disable(self):
             }
         ]
 
-        set_credit_requirements(self.course_key, requirements)
-        old_requirements = CreditRequirement.get_course_requirements(self.course_key)
-        self.assertEqual(len(old_requirements), 1)
 
-        requirements = [
-            {
-                "namespace": "reverification",
-                "name": "i4x://edX/DemoX/edx-reverification-block/assessment_uuid",
-                "display_name": "Assessment 1",
-                "criteria": {}
-            }
-        ]
-        requirements_to_disabled = _get_requirements_to_disable(old_requirements, requirements)
-        self.assertEqual(len(requirements_to_disabled), 1)
-        self.assertEqual(requirements_to_disabled[0], old_requirements[0].id)
+@ddt.ddt
+class CreditProviderIntegrationApiTests(CreditApiTestBase):
+    """
+    Test Python API for credit provider integration.
+    """
 
-        requirements = [
-            {
-                "namespace": "grade",
-                "name": "grade",
-                "display_name": "Grade",
-                "criteria": {
-                    "min_grade": 0.8
-                }
-            },
-            {
-                "namespace": "reverification",
-                "name": "i4x://edX/DemoX/edx-reverification-block/assessment_uuid",
-                "display_name": "Assessment 1",
-                "criteria": {}
-            }
-        ]
-        requirements_to_disabled = _get_requirements_to_disable(old_requirements, requirements)
-        self.assertEqual(len(requirements_to_disabled), 0)
+    USER_INFO = {
+        "username": "bob",
+        "email": "bob@example.com",
+        "full_name": "Bob",
+        "mailing_address": "123 Fake Street, Cambridge MA",
+        "country": "US",
+    }
 
-    def add_credit_course(self, enabled=True):
-        """
-        Mark the course as a credit.
-        """
-        credit_course = CreditCourse(course_key=self.course_key, enabled=enabled)
-        credit_course.save()
-        return credit_course
+    FINAL_GRADE = 0.95
+
+    def setUp(self):
+        super(CreditProviderIntegrationApiTests, self).setUp()
+        self.user = UserFactory(
+            username=self.USER_INFO['username'],
+            email=self.USER_INFO['email'],
+        )
+
+        self.user.profile.name = self.USER_INFO['full_name']
+        self.user.profile.mailing_address = self.USER_INFO['mailing_address']
+        self.user.profile.country = self.USER_INFO['country']
+        self.user.profile.save()
+
+    def test_credit_request(self):
+        self._configure_credit()
```

Comment:
This seems to be used in all the tests. It should be in `setUp()`. Hmm...not sure what to do about `test_user_is_not_eligible` aside from deleting all related objects and calling `_configure_credit()` with the appropriate status..